functions/concepts/src/main/java/com/example/functions/concepts/LazyFields.java

```diff
@@ -30,20 +30,28 @@
   // Always initialized (at cold-start)
   // Warning: Class variables used in Servlet classes must be thread-safe,
   // or else might introduce race conditions in your code.
-  private static final int nonLazyGlobal = fileWideComputation();
+  private static final int NON_LAZY_GLOBAL = fileWideComputation();
+
   // Declared at cold-start, but only initialized if/when the function executes
-  private static Integer lazyGlobal = null;
+  // Uses the "initialization-on-demand holder" idiom
+  // More information: https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom
+  private static class LazyGlobalHolder {
+    // This value is initialized only if (and when) the getLazyGlobal() function below is called
+    private static final Integer LAZY_GLOBAL = functionSpecificComputation();
+  }
+
+  private static Integer getLazyGlobal() {
+    return LazyGlobalHolder.LAZY_GLOBAL;
+  }
 
   @Override
   public void service(HttpRequest request, HttpResponse response)
       throws IOException {
-    // This value is initialized only if (and when) the function is called
-    if (lazyGlobal == null) {
-      lazyGlobal = functionSpecificComputation();
-    }
+    Integer lazyGlobal = getLazyGlobal();
```

Comment:
this should be `LazyGlobalHolder.getInstance()`

Reply:
Thanks - done.